zookeeper-client/zookeeper-client-c/src/mt_adaptor.c

```diff
@@ -479,8 +480,9 @@ void *do_completion(void *v)
         pthread_mutex_unlock(&zh->completions_to_process.lock);
         process_completions(zh);
     }
-    api_epilog(zh, 0);    
-    LOG_DEBUG(LOGCALLBACK(zh), "completion thread terminated");
```

Comment:
Is it more conventional to swap the two statements ? I saw several `return api_epilog(zh, rc)` patterns. Also, line 460 has same problem.

Reply:
Indeed, it would be conventional to log first and then `api_epilog()`.

But `api_epilog()` does `zookeeper_close()` when the rc is zero, i.e. the completion thread is still running after the "terminated" output, which could lead to confusion?